msal/samples/desktop/SampleApp/MainForm.cs

```diff
@@ -58,19 +53,21 @@ public MainForm()
             }
         }
 
-        private async void TabControl1_SelectedIndexChanged(object sender, EventArgs e)
+        // Return void because this is an event handler
+        private async void TabControl1_SelectedIndexChangedAsync(object sender, EventArgs e)
```

Comment:
>private async void TabControl1_SelectedIndexChangedAsync(object sender, EventArgs e) [](start = 8, length = 84)

these are event ui handlers and should probably just have been added to the ignore list. It's like the Main method in a console app. (doc for async will state something similar using better words)

Reply:
(same issue in a few places, but most of the places are spot on, thanks a ton for doing this!)

---
In reply to: [202498640](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1132#discussion_r202498640) [](ancestors = 202498640)

Reply:
i fixed all the event handlers, but you'll have to show me how to get it to ignore those methods so we can remove `async` in them too. they don't show up in MSBuild, just in VS as warnings.